std/traits.d

```diff
@@ -7029,3 +7030,41 @@ template isFinal(X...) if (X.length == 1)
     static assert(!isFinal!(C.sf));
     static assert( isFinal!(C.ff));
 }
+
+/++
+ + Determines whether the type `S` can be copied.
+ + If a type cannot be copied, then code such as `MyStruct x; auto y = x;` will fail to compile.
+ + Copying can be disabled by using `@disable this(this)` for structs.
+ +
+ + Params:
+ +  S = The value type (struct, int, etc.) to check.
+ +
+ + Returns:
+ +  `true` if `S` can be copied.
+ + ++/
+template isCopyable(S)
```

Comment:
```d
enum isCopyable(S) = is(typeof(
	{ S original; S copy = original; }
));
```

No need to restrict for classes and interfaces - references to those are copyable.